### ISSUE TITLE
[Sheet] Add resizable functionality

### ISSUE THREAD

**Issue by macandcheese** [Esri/calcite-design-system#7591]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

Follow-up issue to add resizable functionality to Sheet.
Should coincide with / follow https://github.com/Esri/calcite-design-system/issues/5652
May require updates to the default min / max width css var set here, Shell Panel, for consistency.

### Acceptance Criteria

Resizable functionality is added to Sheet.

### Relevant Info

cc @driskull 

### Which Component

Sheet

### Priority impact

p4 - not time sensitive

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react

### Esri team

Calcite (design)

**Comment by macandcheese** [comment]
+1 from BA cc @ddred11

**Comment by geospatialem** [comment]
This effort should either be combined or added after #5652 to add `shell-panel` improvements to the `resizable` functionality, for the drag handle addordance and to support resizing for when the `displayMode` is `"float"`.

**Comment by SkyeSeitz** [comment]
Final specs [available here](https://www.figma.com/file/A7Z8XlYSkv5FSGwIJGoNu3/%5BShell-Panel%2C-Sheet%2C-Table%5D-%E2%80%9Cresizable%E2%80%9D-%5B7591%2C-5652%5D?type=design&node-id=1%3A1435&mode=dev), sharing the behavior of the resize affordance enhancements to Shell Panel.

**Comment by driskull** [comment]
@SkyeSeitz @ashetland do we need an element drag handle in the designs? I'm just wondering if there is a good enough reason to have a resize element which would take up more space? Even in Figma the Figma side panels are resizable but there isn't a resize handle. My concern is taking more space away from content and if it is worth it. I just want to make sure there is a need for this.

For dialog, we do not have resize handles even when a dialog is resizable. It would probably look pretty odd with them on all sides as well.

**Comment by driskull** [comment]
@SkyeSeitz @ashetland is there an issue for tracking the new drag icon shown in Figma? Will that icon be available for the oct release or do we want to either push this or roll without the icon for this release?

**Comment by SkyeSeitz** [comment]
A request has been submitted and the hope is it will be ready for the release 🤞

**Comment by DitwanP** [comment]
🍡 Verified locally on `dev`
